From a7b5b3a4a61d6547b6d117b452590c2c98580571 Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Thu, 12 Nov 2015 15:40:00 -0800
Subject: [PATCH] nexus/build/os/linuxkernel/os_flags.inc: add -nostdinc to
 NEXUS_CFLAGS for ARM

Reduce chances of accidentally building kernel space code with
user space header files. Change is for ARM only (MIPS builds seem
to do something similar already, but need review).
---
 nexus/build/os/linuxkernel/os_flags.inc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/nexus/build/os/linuxkernel/os_flags.inc b/nexus/build/os/linuxkernel/os_flags.inc
index 9919987..8af058a 100644
--- a/nexus/build/os/linuxkernel/os_flags.inc
+++ b/nexus/build/os/linuxkernel/os_flags.inc
@@ -203,6 +203,9 @@ NEXUS_KERNEL_CFLAGS += \
         -I$(LINUX_OUT)/include/generated/uapi
 endif
 
+NOSTDINC_FLAGS := -nostdinc -isystem $(shell $(CC) -print-file-name=include)
+NEXUS_CFLAGS += $(NOSTDINC_FLAGS)
+
 # When building code for nexus.ko, it seems that some compiler command lines
 # include both NEXUS_CFLAGS and NEXUS_KERNEL_CFLAGS, but some include
 # NEXUS_CFLAGS only. Not including the kernel include paths means either that
-- 
1.9.1

